src/policy/concrete.rs

```diff
@@ -57,10 +57,55 @@ pub enum Policy<Pk: MiniscriptKey> {
     Threshold(usize, Vec<Policy<Pk>>),
 }
 
+/// Detailed Error type for Policies
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+pub enum PolicyError {
+    /// `And` fragments only support two args
+    NonBinaryArgAnd,
+    /// `Or` fragments only support two args
+    NonBinaryArgOr,
+    /// `Thresh` fragment can only have `1<=k<=n`
+    IncorrectThresh,
+    /// `older` or `after` fragment can only have `n >= 0`
+    ZeroTime,
+    /// `after` fragment can only have ` n < 2^31`
+    TimeToofar,
+}
+
+impl error::Error for PolicyError {
+    fn description(&self) -> &str {
+        ""
+    }
+    fn cause(&self) -> Option<&error::Error> {
+        None
+    }
+}
+
+impl fmt::Display for PolicyError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            PolicyError::NonBinaryArgAnd => {
+                f.write_str("And policy fragment must take 2 arguments")
+            }
+            PolicyError::NonBinaryArgOr => f.write_str("Or policy fragment must take 2 arguments"),
+            PolicyError::IncorrectThresh => {
+                f.write_str("Threshold k must be greater than 0 and less than n")
+            }
+            PolicyError::TimeToofar => {
+                f.write_str("Relative/Absolute time must be less than 2^31; n < 2^31")
+            }
+            PolicyError::ZeroTime => {
+                f.write_str("Relative/Absolute time must be greater than 0; n>=0")
```

Comment:
Should be `n  > 0`?

Nit: Could replace "Relative/Absolute time must be …" with "Time must be …". Applies above too.